boards/PDM/Src/Io/Io_Aux1Aux2Efuse.c

```diff
@@ -0,0 +1,131 @@
+#include "Io_Aux1Aux2Efuse.h"
+#include "configs/Io_EfuseConfig.h"
+
+static struct Efuse_Context *aux1_aux2_efuse;
+
+void Io_Aux1Aux2Efuse_Init(SPI_HandleTypeDef *const hspi)
+{
+    assert(hspi != NULL);
+
+    aux1_aux2_efuse = Io_Efuse_Create(
+        hspi, CSB_AUX1_AUX2_GPIO_Port, CSB_AUX1_AUX2_Pin,
+        FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin, FSB_AUX1_AUX2_GPIO_Port,
+        FSB_AUX1_AUX2_Pin, PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, PIN_AUX2_GPIO_Port,
+        PIN_AUX2_Pin);
+}
+
+void Io_Aux1Aux2Efuse_EnableAux1(void)
+{
+    Io_Efuse_Channel0Enable(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DisableAux1(void)
+{
+    Io_Efuse_Channel0Disable(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_EnableAux2(void)
+{
+    Io_Efuse_Channel1Enable(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DisableAux2(void)
+{
+    Io_Efuse_Channel1Disable(aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetStatus(enum Efuse_Status *status)
+{
+    return Io_Efuse_ReadRegister(
+        SO_STATR_ADDR, (uint16_t *)status, aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetAux1Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_ReadRegister(
+        SO_FAULTR_0_ADDR, (uint16_t *)fault, aux1_aux2_efuse);
+}
+
+ExitCode Io_Aux1Aux2Efuse_GetAux2Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_ReadRegister(
+        SO_FAULTR_1_ADDR, (uint16_t *)fault, aux1_aux2_efuse);
+}
+
+bool Io_Aux1Aux2Efuse_IsInFaultMode(void)
+{
+    return Io_Efuse_IsEfuseInFaultMode(aux1_aux2_efuse);
+}
+
+bool Io_Aux1Aux2Efuse_IsInFailSafeMode(void)
+{
+    return Io_Efuse_IsEfuseInFailSafeMode(aux1_aux2_efuse);
+}
+
+void Io_Aux1Aux2Efuse_DelatchFaults(void)
+{
+    Io_Efuse_DelatchFaults(aux1_aux2_efuse);
+}
+
+float Io_Aux1Aux2Efuse_GetAux1Current(void)
+{
+    if (Io_Efuse_ConfigureChannelMonitoring(
+            AUX1_CURRENT_SENSE_CHANNEL, aux1_aux2_efuse) != EXIT_CODE_OK)
+    {
+        return NAN;
```

Comment:
Can you add a comment as to why we'd return a NAN here

Reply:
added